### PR TITLE
Clean up and complete libp2p ping

### PR DESCRIPTION
**Pull request by daviddias**
I was going to add this to js-libp2p, but found out that currently the implementation is broken as it only supports one ping (problem with using always pull.values). Fixing it.

//cc @xicombd

### CODE REVIEW

**Comment by daviddias** [comment]
Done, needs Review

//cc @dignifiedquire @VictorBjelkholm @xicombd

File: package.json [hunk 1]
```diff
@@ -1,52 +1,52 @@
...
     "IPFS"
   ],
   "author": "David Dias <daviddias@ipfs.io>",
   "license": "MIT",
   "bugs": {
-    "url": "https://github.com/diasdavid/js-ipfs-ping/issues"
+    "url": "https://github.com/libp2p/js-ipfs-ping/issues"
   },
-  "homepage": "https://github.com/diasdavid/js-ipfs-ping",
+  "homepage": "https://github.com/libp2p/js-ipfs-ping",
```
Comment (dignifiedquire) [inline package.json]: still the wrong name, should be `js-libp2p-ping`

File: src/config.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+'use strict'
```
Comment (dignifiedquire) [inline src/config.js]: `constants` would be a more appropriate name, given that these are not meant to be changed

File: src/handler.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+function mount (swarm) {
+  swarm.handle(PROTOCOL, (protocol, conn) => {
+    const stream = handshake({ timeout: 0 })
+    const shake = stream.handshake
+
+    // receive and echo back
+    function next () {
+      shake.read(PING_LENGTH, (err, buf) => {
+        if (err) {
+          return log.error(err)
```
Comment (dignifiedquire) [inline src/handler.js:26]: `err` true shouldn't be logged as it is, as it means regular stream end, not a failure. So this should be at least 

```js
if (err === true) {
  return log.error("Stream ended prematurely")
}

if (err) {
  return log.error(err)
}
```

File: src/handler.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        }
+
+        shake.write(buf)
+        next()
```
Comment (dignifiedquire) [inline src/handler.js]: this is recursive, how will this ever stop?

File: src/handler.js (hunk 3 above)
Comment (dignifiedquire) [reply src/handler.js:26]: actually it looks like the stop happens, such that the dialer closes, so `err === true` should be just `return` from the looks

File: src/util.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+'use strict'
+
+const randomBytes = require('random-bytes')
```
Comment (dignifiedquire) [inline src/util.js]: hmm why do you need a module for random bytes? there is `crypto.getRandomValues()` in the browser and `crypto.randomBytes()` in node.js

File: test/test-ping.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    parallel([
+      (cb) => swarmA.close(cb),
+      (cb) => swarmB.close(cb)
+    ], done)
+  })
+
+  it('mount ping protocol', () => {
+    Ping.mount(swarmA)
+    Ping.mount(swarmB)
+  })
```
Comment (dignifiedquire) [inline test/test-ping.js]: please make this part of the `before`, as other wise calling `.only` on the following tests will always fail

**Review by dignifiedquire — CHANGES_REQUESTED** [review]

File: src/util.js (hunk 5 above)
Comment (daviddias) [reply src/util.js]: it is what random-bytes uses underneath

What about we expose this feature in libp2p-crypto?

**Comment by daviddias** [comment]
applied CR

File: src/util.js (hunk 5 above)
Comment (dignifiedquire) [reply src/util.js]: sounds good

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**